### ISSUE TITLE
[from:policy] Publish Laser reduced encoder features

### ISSUE THREAD

**Issue by kaihe** [kaihe/contra_nes_data#15]
## Summary
Publish the frozen static encoder's reduced `4x4x256` Laser features so policy can tune the projection without rerunning the convolutional trunk.

## Done / request
- [x] Raw Laser D10k RGB release is complete: 10,293 episodes / 1,075,404 frames.
- [ ] Encode every frame through `view_backbone` and `reduce`, stopping before `proj`.
- [ ] Store float16 `(T,256,4,4)` features plus raw actions and episode metadata.
- [ ] Record encoder checkpoint SHA-256, feature boundary/version, shape, dtype, and action alignment in the spec.

## Paths
- Encoder checkpoint: `/home/kaihe/code/contra_nes_policy/runs/encoder/2026-07-31/18-00-11/checkpoints/encoder-final.pt`
- Raw frames: `/home/kaihe/code/contra_nes_data/game_trace/datahouse/level1/boss/laser/frames/`
- Policy experiment: `/home/kaihe/code/contra_nes_policy/doc/0024-exp-unfrozen-laser-encoder.md`

## Next action (for this repo)
1. Add a versioned reduced-feature representation beside token and frame shards.
2. Publish the complete Laser D10k fingerprint membership and verify sampled features against a live encoder forward.
3. Notify policy with the catalog selector and measured storage size.

## Must not break
- Use the exact frozen trunk through `reduce`; do not include `proj` or `token_ln`.
- Preserve `frames[i]` post-action and actions unshifted; policy applies target `actions[i+1]`.
- Do not alter existing token/frame shards or their catalog identities.

## Related
- `kaihe/contra_nes_policy#22`
- Policy `doc/0024-exp-unfrozen-laser-encoder.md`
- ROCKET-2 analogue: frozen view backbone plus trainable projection/resampler.

**Comment by kaihe** [comment]
Policy-side contract clarified:

- The data representation stops exactly after the frozen encoder `reduce` block: float16 `(T, 256, 4, 4)`.
- Do **not** apply or serialize outputs from `proj` or `token_ln`; those modules are now owned by the policy checkpoint.
- Policy will initialize `proj` (`4096 -> 512 -> 512`) and `token_ln` from the existing pretrained encoder, then train them with the L core under one `1e-4` learning rate.
- The matched control will load the same reduced features but keep the pretrained projection frozen.

Please include enough provenance in the representation spec to rebuild the exact frozen producer: encoder checkpoint SHA-256, boundary name/version (`view_backbone + reduce`), tensor shape/order, dtype, input resize contract, and sampled live-forward verification.

**Comment by kaihe** [comment]
Implemented and published on `feat/laser-reduced-features` at `a758342`.

Policy selector:
```json
{"level":1,"task":"boss","weapon":"laser","representation":"reduced-view-v1","encoder_sha256":"f36041bc69f1ce20781d5200bc89970b1b305e12bff5ae826b23581ca0f1923c"}
```

Release spec:
`/home/kaihe/code/contra_nes_data/game_trace/datahouse/level1/boss/laser/features/reduced-view-v1/f36041bc69f1ce20781d5200bc89970b1b305e12bff5ae826b23581ca0f1923c/spec.json`

Audit:
- 42 shards, 10,293 episodes, 1,075,404 frames
- 8,856,750,080 bytes
- ordered membership SHA-256 `616126422ac17bcfe3d07fec9cd329533d616d08e80c8bd65565932b492184e3`
- exact frame/feature membership: 10,293 / 10,293, zero missing
- live-forward sample: 9 episodes / 27 frames, max absolute float16 delta `0.00390625` under `0.0078125` tolerance
- `proj` and `token_ln` excluded; actions retained unshifted
- full data-repo suite: 120 passed

Contract and measured result are in `doc/0022-design-laser-reduced-features.md`.
